### PR TITLE
Initial Sawtooth Whitepaper

### PR DESCRIPTION
**Pull request by ineffectualproperty**
Signed-off-by: Kelly Olson <ineffectualproperty@users.noreply.github.com>

### CODE REVIEW

**Comment by vaporos** [comment]
Please also wrap at <80 columns.

File: whitepaper.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+* **PoET-Simulator**: Sawtooth includes an implementation of PoET that simulates the secure instructions. This mechanism enables large-scale consensus populations, but forgoes Byzantine fault tolerance.
+
+* **RAFT** (Under Development): The RAFT consensus protocol is a crash-fault tolerant ‘voting’-style consensus algorithm. It is designed for high throughput, low latency transactions. 
+
+**Privacy and Confidentiality**
+
+Blockchain privacy and confidentiality is one of the key challenges to blockchain adoption in the enterprise. To-date there have been two general means of providing privacy on a blockchain:
+
+1. **Separation of Concerns **- Data is only sent to the relevant parties of a transaction, with an optional hash of that data broadcast to all blockchain nodes. This means of privacy has been implemented by R3’s Corda, JPMorgan’s Quorom, and Hyperledger Fabric through it’s channels model. While this method achieves the privacy and data locality concerns, it has a few drawbacks. The key disadvantage of this model is that it creates a number of ‘sub-chains’ and transferring assets between these chains adds significant complexity or reduces confidentiality.
```
[the remark this answers is not in the document]
Comment (chenette) [reply whitepaper.md]: I feel like "achieves" and "concerns" don't go together ("While this method achieves the privacy and data locality concerns"). I'd suggest changing "achieves" to "addresses", or else "concerns" to something like "goals".

File: whitepaper.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+**Distributed Ledgers**
+
+Since the release of Ethereum, a variety of other ‘distributed ledgers’ implementations have been created to meet the needs of the enterprise. These distributed ledger implementations include software that expands upon existing protocols (e.g. Bitcoin - MultiChain, Chain, Blockstream and Ethereum - Quorum) and creates entirely new implementations (Corda and Fabric). These implementations deviate from public blockchains to address limitations in existing protocols such as throughput, security, efficiency, usability, and confidentiality. Some distributed ledgers differ substantially from their blockchain predecessors as a result of changes to the underlying networking, journalling layer, smart contract, and consensus layers; some even going so far as to not include ‘blocks’ or a ‘chain’.
+
+**Related Technologies**
+
+Peer-to-peer networking
```
Comment (chenette) [inline whitepaper.md]: I feel like this should be a bulleted list.

File: whitepaper.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+Event driven databases
+
+Replicated state machines
+
+Distributed databases
+
+Replicated logs
+
+Consensus algorithms.
```
Comment (chenette) [inline whitepaper.md]: Delete the period

File: whitepaper.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+1. **Separation of Concerns **- Data is only sent to the relevant parties of a transaction, with an optional hash of that data broadcast to all blockchain nodes. This means of privacy has been implemented by R3’s Corda, JPMorgan’s Quorom, and Hyperledger Fabric through it’s channels model. While this method achieves the privacy and data locality concerns, it has a few drawbacks. The key disadvantage of this model is that it creates a number of ‘sub-chains’ and transferring assets between these chains adds significant complexity or reduces confidentiality.
+
+2. **Computation on Encrypted Data** - In this approach, encrypted data is broadcast across the entire network in a way that the data can be computed on. This may entail advanced cryptographic techniques such as homomorphic encryption or zero-knowledge proofs, or leverage a trusted execution environment such as Intel® Software Guard Extensions (Intel® SGX). This method too comes with several drawbacks. Such drawbacks may include the need for trusted hardware or performance and usability limitations with advanced cryptographic techniques.
```
Comment (chenette) [inline whitepaper.md]: Even though it is okay to end a sentence with a preposition, I suggest changing "in a way that the data can be computed on" to something like "in a way that allows computation on the data".

File: whitepaper.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+Currently Sawtooth does not dictate a privacy and confidentiality approach. The Sawtooth team has experimented with several approaches, including trusted computation of transaction off-chain, which can be found here - [https://sawtooth.hyperledger.org/docs/core/releases/0.8.8/examples/private_utxo/overview.html](https://sawtooth.hyperledger.org/docs/core/releases/0.8.8/examples/private_utxo/overview.html).
+
+This area is an active area of research and a key priority for future Sawtooth releases. The Sawtooth team believes that different privacy models are appropriate for different deployments and intends to leverage data isolation, cryptographic, and trusted hardware techniques in their approach.
+
+**Decentralized Governance**
+
+One of the unique features of Sawtooth is the on-chain governance mechanism enabled by the ‘settings’ transaction family. The settings transaction family provides a methodology for storing on-chain configuration settings such as which consensus mechanism is used and which transaction processors are enabled on the network
```
Comment (chenette) [inline whitepaper.md]: Settings should start with a capital S (for all occurrences of "Settings transaction family") to be consistent with the Sawtooth docs.  I would suggest deleting the single quotes, too.

File: whitepaper.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+The secondary goal of Seth is to make it easy to port existing EVM smart contracts, and the DApps that depend on them, to Sawtooth. This has been largely accomplished by replicating the [Ethereum JSON RPC API](https://github.com/ethereum/wiki/wiki/JSON-RPC). This also enables developers to leverage existing Ethereum tooling such as IDEs when developing for Sawtooth.
+
+It is important to note that Seth is not a complete Ethereum implementation. The Sawtooth platform has made fundamental design decisions that differ from those made by the Ethereum platform. 
+
+**Conclusion**
+
+Sawtooth is a blockchain platform designed for enterprise usage. Sawtooth builds upon the learnings of platforms before it such as Bitcoin and Ethereum and is designed with a focus on modularity and extensability. Sawtooth is unique from existing blockchain platforms in that it allows developers to program their smart contracts in a variety of traditional programming languages. Future Sawtooth developments will focus on increasing the throughput and privacy of transactions.
```
Comment (chenette) [inline whitepaper.md]: I'd say "lessons" instead of "learnings".

Comment (chenette) [reply whitepaper.md]: Also, I would put commas before and after "such as Bitcoin and Ethereum".

Comment (chenette) [reply whitepaper.md]: Alas, "unique from" is incorrect (according to us grammar pedants).  My first thought would be to say "distinct from" instead, but I like the the word "unique" here.  Perhaps change this sentence to "Sawtooth is unique among existing blockchain platforms ...".

File: whitepaper.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+**Distributed Log**
+
+One of the foundational components of blockchains and replicated state machines is ensuring that nodes on the network process transactions in a consistent order. On Sawtooth, this ordered list of transactions is assembled into a data structure called a ‘blockchain’. The ‘blockchain’ contains an ordered set of cryptographically linked blocks, beginning with a genesis block. Each block in the linked list contains an ordered set of transactions. Nodes reach consensus on the correct ordering of this log through a consensus algorithm.
+
+Currently the Sawtooth blockchain arrives at a total ordering of every transaction. That is, every transaction is strictly before, or after, any other transaction. One area for exploration is modifying the blockchain structure to allow for partial ordering between non-dependent transactions. On such data structure is a directed acyclic graph. This would potentially allow for the blockchain network to be partitioned or ‘sharded’ to improve scalability and reduce storage requirements.
```
Comment (chenette) [inline whitepaper.md]: Also, I'd add a comma after "Currently".

File: whitepaper.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+Interpreted Languages: Python and Javascript
+
+Compiled Languages: Rust, C++ and Go
+
+Virtual Machines: Ethereum Virtual Machine and Java
+
+One of the fundamental tradeoffs in smart contract language design is between security and expressibility. By creating a domain specific transaction processor, it is much easier to limit the types of actions that can be performed on a blockchain network, which can improve security and performance. Furthermore, by leveraging more mature programming languages, it is possible to leverage existing tools that can perform static, dynamic, and formal analysis of your smart contract code.
+
+At its heart Sawtooth is agnostic to the smart contract transaction processing language. Sawtooth has proven the ability to run Solidity contracts through its integration with the Hyperledger Burrow EVM. The Sawtooth development team intends to expand the programming languages that it supports, and to integrate with existing and emerging smart contract engines such as Chain’s Ivy, and Digital Asset Holding’s DAML.
```
Comment (chenette) [inline whitepaper.md]: I'd add a comma after "At its heart".

Comment (chenette) [reply whitepaper.md]: Also, delete the comma after "Chain's Ivy".

File: whitepaper.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+
+**Decentralized Governance**
+
+One of the unique features of Sawtooth is the on-chain governance mechanism enabled by the ‘settings’ transaction family. The settings transaction family provides a methodology for storing on-chain configuration settings such as which consensus mechanism is used and which transaction processors are enabled on the network
+
+This design of the settings transaction family supports two authorization options, either a single authorized key which can make changes or multiple authorized keys. In the case of multiple keys, a percentage of votes signed by the keys is required before a change is made.
+
+**Sawtooth and Ethereum Compatibility**
+
+The primary goal of the Sawtooth-Ethereum integration project, affectionately dubbed "Seth", is to add support for running Ethereum Virtual Machine smart contracts to the Hyperledger Sawtooth platform. In order to make this possible, the Hyperledger Sawtooth project worked with the [Hyperledger Burrow](https://github.com/hyperledger/burrow) project to integrate their EVM implementation, the Burrow EVM, into the Hyperledger Sawtooth platform.
```
Comment (grkvlt) [inline whitepaper.md]: Change introductory sentence to read as follows, with link to `sawtooth-seth` repository:

> The primary goal of the Sawtooth-Ethereum integration project [Seth](https://github.com/hyperledger/sawtooth-seth/) is to ...

Comment (grkvlt) [inline whitepaper.md]: Remove phrase `the Burrow EVM` from the second sentence.

**Comment by ineffectualproperty** [comment]
Made changes, will post to sawtooth-website repo when created

**Pull request closed without merging**